desktop/src/app.rs

```diff
@@ -218,64 +214,74 @@ impl App {
+                                        self.gui
+                                            .lock()
+                                            .expect("Gui lock")
+                                            .show_context_menu(context_menu);
                                     }
-                                    ElementState::Released => PlayerEvent::MouseUp { x, y, button },
-                                };
-                                if state == ElementState::Pressed
-                                    && button == RuffleMouseButton::Right
-                                {
-                                    // Show context menu.
-                                    // TODO: Should be squelched if player consumes the right click event.
-                                    let context_menu = player.prepare_context_menu();
-                                    self.gui
-                                        .lock()
-                                        .expect("Gui lock")
-                                        .show_context_menu(context_menu);
+                                    player.handle_event(event);
                                 }
-                                player.handle_event(event);
                             }
                             check_redraw = true;
                         }
                         WindowEvent::MouseWheel { delta, .. } => {
                             use ruffle_core::events::MouseWheelDelta;
                             use winit::event::MouseScrollDelta;
                             if let Some(mut player) = self.player.get() {
-                                let delta = match delta {
-                                    MouseScrollDelta::LineDelta(_, dy) => {
-                                        MouseWheelDelta::Lines(dy.into())
-                                    }
-                                    MouseScrollDelta::PixelDelta(pos) => {
-                                        MouseWheelDelta::Pixels(pos.y)
-                                    }
-                                };
-                                let event = PlayerEvent::MouseWheel { delta };
-                                player.handle_event(event);
+                                if player.is_playing() {
+                                    let delta = match delta {
+                                        MouseScrollDelta::LineDelta(_, dy) => {
+                                            MouseWheelDelta::Lines(dy.into())
+                                        }
+                                        MouseScrollDelta::PixelDelta(pos) => {
+                                            MouseWheelDelta::Pixels(pos.y)
+                                        }
+                                    };
+                                    let event = PlayerEvent::MouseWheel { delta };
+                                    player.handle_event(event);
```

Comment:
could we instead wrap handle_event in an is-playing-checking function?

Reply:
I'll take a look! It won't handle all of the cases but maybe it'll help